src/rollback_to_stable/rts_btree.c

```diff
@@ -600,7 +600,10 @@ __rts_btree_ondisk_fixup_key(WT_SESSION_IMPL *session, WT_REF *ref, WT_ROW *rip,
     if (hs_cursor != NULL)
         WT_TRET(hs_cursor->close(hs_cursor));
     if (dryrun) {
-        WT_ASSERT(session, !valid_update_found || upd == NULL);
+        /*
+         * Dry runs don't modify the database so any upd structure allocated by this function is not
+         * in use and must be cleaned up.
+         */
         __wt_free_update_list(session, &upd);
```

Comment:
The braces around both changed sections are now unnecessary.

Reply:
I've updated the second section but left the first as all if branches contain multiple statements.